x-pack/plugins/index_management/common/types/templates.ts

```diff
@@ -42,6 +43,7 @@ export interface TemplateDeserialized {
     aliases?: Aliases;
     mappings?: Mappings;
   };
+  deprecated?: boolean;
```

Comment:
Just FMI: Why are we adding this `deprecated` flag for index templates? Is it somehow needed for the ILM policy deprecated badge?